netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -193,23 +195,27 @@ public void post(final String text) throws IOException {
 
     @Override
     public long unread() throws IOException {
-        final Item item = this.region.table(DyFriends.TBL)
-            .frame()
-            .through(
-                new QueryValve()
-                    .withLimit(1)
-                    .withAttributesToGet(DyFriends.ATTR_UNREAD)
-            )
-            .where(DyFriends.HASH, Conditions.equalTo(this.bout))
-            .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
-        final long unread;
-        if (item.has(DyFriends.ATTR_UNREAD)) {
-            unread = Long.parseLong(item.get(DyFriends.ATTR_UNREAD).getN());
-        } else {
-            unread = 0L;
+        try {
+            final Item item = this.region.table(DyFriends.TBL)
+                .frame()
+                .through(
+                    new QueryValve()
+                        .withLimit(1)
+                        .withAttributesToGet(DyFriends.ATTR_UNREAD)
+                )
+                .where(DyFriends.HASH, Conditions.equalTo(this.bout))
+                .where(DyFriends.RANGE, Conditions.equalTo(this.self))
+                .iterator().next();
+            final long unread;
+            if (item.has(DyFriends.ATTR_UNREAD)) {
+                unread = Long.parseLong(item.get(DyFriends.ATTR_UNREAD).getN());
+            } else {
+                unread = 0L;
+            }
+            return unread;
+        } catch (final NoSuchElementException ex) {
+            throw new RsFailure(ex);
```

Comment:
@2686747 I don't think this should be a totally nondescript `RsFailure` here either way, especially after removing the exception.
This should simply be the same error you'd get in case the bout hadn't existed in the first place.